parlai/core/agents.py

```diff
@@ -154,6 +156,78 @@ def shutdown(self):
         """
         pass
 
+    def self_observe(self, self_message: Message) -> None:
+        """
+        Observe one's own utterance.
+
+        :param self_message:
+            The message corresponding to the output from batch_act.
+        """
+        pass
+
+    def respond(
+        self, text_or_message: Union[str, Message], **other_message_fields
+    ) -> str:
+        """
+        An agent convenience function which calls the act() and provides a string
+        response to a text or message field.
+
+        :param Union[str, Message] text_or_message:
+            A string for the 'text' field or a message which MUST
+            comprise of the 'text' field apart from other fields.
+        :param kwargs other_message_fields:
+            Provide fields for the message in the form of keyowrd arguments.
+        :return:
+            Agent's response to the message.
+        :rtype:
+            str
+        """
+        if isinstance(text_or_message, str):
+            observation = Message(text=text_or_message, **other_message_fields)
+        else:
+            observation = Message(**text_or_message, **other_message_fields)
+            if 'text' not in observation:
+                raise RuntimeError('The agent needs a \'text\' field in the message.')
+
+        if 'episode_done' not in observation:
+            observation['episode_done'] = True
+        agent = self.clone()
+        agent.observe(observation)
+        response = agent.act()
+        return response['text']
+
+    def batch_respond(self, messages: List[Message]) -> List[str]:
```

Comment:
Ditto here, should this be `List[Message]`?

Reply:
Again, could you take a look at the function signatures in https://github.com/facebookresearch/ParlAI/issues/3408?